src/Descriptor.hpp

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include "merc.h"
+
+#include <list>
+#include <optional>
+#include <string>
+#include <string_view>
+
+// Connected state for a descriptor.
+enum class DescriptorState {
+    Playing = 0,
+    GetName = 1,
+    GetOldPassword = 2,
+    ConfirmNewName = 3,
+    GetNewPassword = 4,
+    ConfirmNewPassword = 5,
+    GetNewRace = 6,
+    GetNewSex = 7,
+    GetNewClass = 8,
+    GetAlignment = 9,
+    DefaultChoice = 10,
+    GenGroups = 11,
+    ReadIMotd = 13,
+    ReadMotd = 14,
+    BreakConnect = 15,
+    GetAnsi = 16,
+    CircumventPassword = 18, // used by doorman
+    Disconnecting = 254, // disconnecting having been playing
+    DisconnectingNp = 255, // disconnecting before playing
+};
+
+const char *short_name_of(DescriptorState state);
+const char *name_of(DescriptorState state);
+
+/*
+ * Descriptor (channel) structure.
+ */
+struct Descriptor {
+    static constexpr size_t MaxInbufBacklog = 50u;
+    std::list<std::string> pending_commands_;
+    std::string last_command_;
+
+    [[nodiscard]] std::optional<std::string> pop_raw();
+
+public:
+    Descriptor *next{};
+    Descriptor *snoop_by{};
```

Comment:
Dunno if we're doing TODOs, but if so, maybe having a couple of them here would be good. IMO it'd be good to:
* Switch these fields to 'private'.
* Add a trailing _ (IIRC that was the style we ended up agreeing on, right?)

Reply:
Yes! all these are planned! Just one small step at a time (see the upcoming PR that does some of these).  :D